### PR TITLE
chore(TDI-37688): add logging for data manipulating.

### PR DESCRIPTION
**Pull request by mbasiuk-talend**
**What is the current behavior?** (You can also link to an open issue here)
No logging info about rejected/successful flow on data.

**What is the new behavior?**
Add logging.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [x] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceWriter.java [hunk 1]
```diff
@@ -176,8 +177,9 @@ public void write(Object datum) throws IOException {
                     .createIndexedRecordConverter(datum.getClass());
         }
         IndexedRecord input = factory.convertToAvro(datum);
-
-        switch (sprops.outputAction.getValue()) {
+        OutputAction outputAction = sprops.outputAction.getValue();
+        LOGGER.info("Prepare to {} record {}.", outputAction.name().toLowerCase(), dataCount);
```
Comment (dmytro-sylaiev) [inline components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceWriter.java]: I suppose I18N is needed here

File: components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceWriter.java [hunk 2]
```diff
@@ -475,6 +481,7 @@ private void handleSuccess(IndexedRecord input, String id, String status) {
             }
             successfulWrites.add(successful);
         }
+        LOGGER.info("Has {} record {} successfully.", getPastForm(sprops.outputAction.getValue()), dataCount);
```
Comment (dmytro-sylaiev) [inline components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceWriter.java]: Also I18N

File: components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceWriter.java [hunk 3]
```diff
@@ -525,13 +533,15 @@ else if (TSalesforceOutputProperties.FIELD_ERROR_FIELDS.equals(outField.name()))
...
-                        } else if (TSalesforceOutputProperties.FIELD_ERROR_MESSAGE.equals(outField.name()))
+                        } else if (TSalesforceOutputProperties.FIELD_ERROR_MESSAGE.equals(outField.name())) {
                             outValue = error.getMessage();
+                        }
                     }
                     reject.put(outField.pos(), outValue);
                 }
                 rejectedWrites.add(reject);
             }
+            LOGGER.info("Failed to {} record {}.", sprops.outputAction.getValue().name().toLowerCase(), dataCount);
```
Comment (dmytro-sylaiev) [inline components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceWriter.java]: Also I18N is needed here

File: components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceWriter.java [hunk 4]
```diff
@@ -593,6 +603,10 @@ else if (TSalesforceOutputProperties.FIELD_ERROR_FIELDS.equals(outField.name()))
     @Override
     public Result close() throws IOException {
         logout();
+        LOGGER.info("Has " + getPastForm(sprops.outputAction.getValue()) + successCount + " record" + (successCount == 1 ? "" : "s")
```
Comment (dmytro-sylaiev) [inline components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceWriter.java]: Also I18N is needed here

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/890/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/890/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/891/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/891/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/892/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/892/)

**Pull request merged**